### PR TITLE
fix: resolve conflict import paths in the generated files

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

#44 

#### 2. What is the scope of this PR (e.g. component or file name):

pkg/swagger/generator/model.go
pkg/swagger/generator/structs.go
pkg/swagger/generator/templates/header.gotmpl

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

resolve conflict import paths when there are different import paths with the same package name

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [X] Manual test (add detailed scripts or steps below)
- [ ] Other

1. first do the pre-process 
2. kcl-openapi generate model -f preprocessed-swagger.json
3. check the import stmts in `io/k8s/api/apps/v1/deployment_spec.k`

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
resolve conflict import paths in the generated files
```

### CODE REVIEW

File: pkg/swagger/generator/model.go [hunk 1]
```diff
@@ -161,73 +155,96 @@ func makeGenDefinitionHierarchy(name, pkg, container string, schema spec.Schema,
...
+			if v.AsName == asName {
+				conflict = true
+				break
 			}
 		}
+		if !conflict {
+			return asName
+		}
 	}
+	return "kusionMangled" + strings.ToTitle(asName)
```
Comment (Peefy) [inline pkg/swagger/generator/model.go:200]: 1. Why use `"kusionMangled"`?
2. What would happen if there were multiple conflicts?

Comment (amyXia1994) [reply pkg/swagger/generator/model.go]: 1. use `import as` to resolve possible import conflicts like `import io.k8s.apimachinery.pkg.apis.meta.v1` and `import io.k8s.api.core.v1`
2. concat parent package name as the prefix to resolve conflicts: `metaV1` and `coreV1`; If conflicts remain, continue concat until <a>. no conflict <b>. no parent package exists
3. If all the parent package name is added as the prefix and conflicts remain, a `kusionMangled` prefix will be added to resolve it. The `kusionMangled` prefix is just a fallback.
4. After all these are processed, if conflicts remain(which should not happen for well-designed APIs), we'll print a warning message and live the conflict for users to properly resolve.

File: pkg/swagger/generator/model.go [hunk 2]
```diff
@@ -161,73 +155,102 @@ func makeGenDefinitionHierarchy(name, pkg, container string, schema spec.Schema,
...
+	mangledAsName := "kusionMangled" + strings.ToTitle(asName)
+	for _, v := range imp {
+		if v.AsName == asName {
+			log.Printf("[WARN] the import paths in module %s.%s are confict, please resolve it properly", sch.Pkg, sch.Module)
 		}
+	}
+	return mangledAsName
+}
+
+// findImports add import stmt from sch to opts if needed to the map of import stmts.
```
Comment (Peefy) [inline pkg/swagger/generator/model.go:209]: s/findImports/collectImports

Besides, the comments of the function does not seem to match the usage of the function.

Comment (amyXia1994) [reply pkg/swagger/generator/model.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**